openedx/core/djangoapps/xblock/runtime/blockstore_field_data.py

```diff
@@ -255,3 +256,92 @@ def free_unused_definitions(self):
+    BlockstoreFieldData can only store field data that is not usage-specific. So
+    we store data about the <xblock-include /> elements that define the children
+    in BlockstoreFieldData (since that is not usage-specific), and this field
+    data implementation loads that <xblock-include /> data and transforms it
+    into the usage keys that comprise the standard .children field.
+    """
+    def __init__(self, blockstore_field_data):
+        """
+        Initialize this BlockstoreChildrenData instance.
+        """
+        # The data store that holds Scope.usage and Scope.definition data:
+        self.authored_data_store = blockstore_field_data
+        super(BlockstoreChildrenData, self).__init__()
+
+    def _check_field(self, block, name):  # pylint: disable=unused-argument
+        """
+        Given a block and the name of one of its fields, check that we will be
+        able to read/write it.
+        """
+        if name != 'children':
+            raise InvalidScopeError("BlockstoreChildrenData can only read/write from a field named 'children'")
+
+    def get(self, block, name):
+        """
+        Get the "children' field value.
+
+        We do this by reading the parsed <xblock-include /> values from
+        the regular authored data store and then transforming them to usage IDs.
+        """
+        self._check_field(block, name)
+        children_includes = self.get_includes(block)
+        if not children_includes:
+            return []
+        # Now the .children field is required to be a list of usage IDs:
+        learning_context = get_learning_context_impl(block.scope_ids.usage_id)
+        child_usages = []
+        for parsed_include in children_includes:
+            child_usages.append(
+                learning_context.usage_for_child_include(
+                    block.scope_ids.usage_id, block.scope_ids.def_id, parsed_include,
+                )
+            )
+        return child_usages
+
+    def set(self, block, name, value):
+        """
+        Set the value of the field; requires name='children'
+        """
+        self._check_field(block, name)
+        children_includes = self.authored_data_store.get(block, CHILDREN_INCLUDES)
+        if len(value) != len(children_includes):
+            raise RuntimeError(
+                "This runtime does not allow changing .children directly - use runtime.add_child_include instead."
+            )
+        # This is a no-op; the value of 'children' is derived from CHILDREN_INCLUDES
+        # so we never write to the children field directly. All we do is make sure it
+        # looks like it's still in sync with CHILDREN_INCLUDES
+
+    def get_includes(self, block):
+        """
+        Get the list of <xblock-include /> elements representing this XBlock's
+        children.
+        """
+        try:
+            return self.authored_data_store.get(block, CHILDREN_INCLUDES)
+        except KeyError:
+            return []
```

Comment:
Can you please add a comment as to why it's okay to return an empty list here instead of erroring?